tools/cpp/logger/main.cpp

```diff
@@ -683,6 +703,7 @@ int main(int argc, char *argv[])
     signal(SIGINT,  sighandler);
     signal(SIGQUIT, sighandler);
     signal(SIGTERM, sighandler);
+    signal(SIGHUP,  sighup_handler);
```

Comment:
You can register this handler only in the cases you want to actually handle the signal. Specifically `args.auto_increment || args.rotate > 0`